Fix part of #4865: Refactor code to use profileId

jainv4156 opened the pull request:
## Explanation
Fixes part of https://github.com/oppia/oppia-android/issues/4865

## Essential Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [ ] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
<!-- Delete these section if this PR does not include UI-related changes. -->
If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing



[Screencast from 2025-03-05 13-48-43.webm](https://github.com/user-attachments/assets/5dc35f47-d858-4c08-a317-0b8f07931ca5)

jainv4156 committed:
Refactor ProfileId provision process to multiple activity

jainv4156 committed:
Delete config/kitkat_main_dex_class_list.txt

jainv4156 committed:
fix lint issues

jainv4156 committed:
Merge remote-tracking branch 'origin/new_branch' into new_branch

jainv4156 committed:
refactor testfiles

jainv4156 committed:
Updated commit without unwanted file

jainv4156 committed:
fix lint errors

jainv4156 committed:
Delete config/kitkat_main_dex_class_list.txt

jainv4156 commented:
@adhiamboperes PTAL,
Thanks!

adhiamboperes commented on app/src/main/java/org/oppia/android/app/player/exploration/ExplorationFragmentPresenter.kt:
```diff
@@ -70,14 +70,15 @@ class ExplorationFragmentPresenter @Inject constructor(
   fun handleViewCreated() {
-    val profileDataProvider = profileManagementController.getProfile(retrieveArguments().profileId)
+    val profileDataProvider = profileManagementController.getProfile(
+      activity.intent.extractCurrentUserProfileId()
```
Use the `profileId` variable created above(you most likely need to make it a global variable first).

jainv4156 replied:
Done

adhiamboperes commented on app/src/main/java/org/oppia/android/app/player/exploration/ExplorationFragmentPresenter.kt:
```diff
@@ -161,7 +162,7 @@ class ExplorationFragmentPresenter @Inject constructor(
         classroomId, topicId, storyId, explorationId
       ),
-      ProfileId.newBuilder().apply { internalId = internalProfileId }.build()
+      activity.intent.extractCurrentUserProfileId()
```
Use the profileId variable as per the comment above.

jainv4156 replied:
Done

adhiamboperes commented on app/src/main/java/org/oppia/android/app/survey/SurveyActivity.kt:
```diff
@@ -48,12 +49,10 @@ class SurveyActivity : InjectableAutoLocalizedAppCompatActivity() {
       explorationId: String
     ): Intent {
       val params = SurveyActivityParams.newBuilder().apply {
-        this.profileId = profileId
```
This has been removed but I do not see a replacement with the decorator here or at the function call site(SurveyWelcomeDialogFragmentPresenter.kt)

adhiamboperes commented on app/src/main/java/org/oppia/android/app/story/StoryActivity.kt:
```diff
@@ -65,7 +64,6 @@ class StoryActivity :
     startActivity(
       ExplorationActivity.createExplorationActivityIntent(
         this,
-        profileId,
```
Here and routeToResumeLesson below, at which point is the intent decorated with the profileId as a replacement?

adhiamboperes commented on app/src/main/java/org/oppia/android/app/topic/TopicActivity.kt:
```diff
@@ -117,7 +116,6 @@ class TopicActivity :
     startActivity(
       ExplorationActivity.createExplorationActivityIntent(
         this,
-        profileId,
```
Here and routeToResumeLesson below, at which point is the intent decorated with the profileId as a replacement?

adhiamboperes reviewed (changes requested):
Thanks @jainv4156!

My main concern here is that some profileId arguments have been removed, but I was not able to see a corresponding `intent.decorateWithCurrentUserProfileId`. This is possible if the decorator was added in a previous PR, but it is best to play through the app manually for the affected pathways to verify that no crashes are happenning, since these are critical user pathways.

jainv4156 committed:
added profile Decorater in required function

jainv4156 committed:
fix import issue

jainv4156 committed:
fix lint issue

jainv4156 committed:
Merge remote-tracking branch 'origin/new_branch' into new_branch

jainv4156 committed:
Merge branch 'develop' into new_branch

jainv4156 commented:
@adhiamboperes PTAL , 
Thanks !

adhiamboperes commented:
> Thanks @jainv4156!
> 
> My main concern here is that some profileId arguments have been removed, but I was not able to see a corresponding `intent.decorateWithCurrentUserProfileId`. This is possible if the decorator was added in a previous PR, but it is best to play through the app manually for the affected pathways to verify that no crashes are happenning, since these are critical user pathways.


@jainv4156, on the open comments, I would like to know if the profileId arguments were required or not, and also, have you tested the app as advised?

jainv4156 commented:
@adhiamboperes, I manually tested the app, and I've added the video above. There’s no crash or any error in the logcat. You can check the video, and if you feel I missed anything in testing, let me know how I can test it better.

About the profileId argument — it’s not really required now. Earlier, it was being passed through the function to different activities and fragments, but I removed that and used the intent.profileId decorator instead, as mentioned in the issue. You can also see the profileId decorator being added in this commit: https://github.com/oppia/oppia-android/pull/5722/commits/6619af86d29786c01ed87727e0a817625893e3fc.

If there's still anything left to fix or improve, just let me know.

adhiamboperes commented on model/src/main/proto/arguments.proto:
```diff
@@ -93,11 +93,9 @@ message DestinationScreen {
 
 // Intent extras for RecentlyPlayedActivity.
 message RecentlyPlayedActivityParams {
```
I found a usage for this proto in the RecentlyPlayedActivityTest. Shouldn't the profileId there be removed as part of the clean up? I can't seem to see it as part of the changed files.

adhiamboperes commented on app/src/test/java/org/oppia/android/app/testing/RecentlyPlayedSpanTest.kt:
```diff
@@ -280,11 +266,10 @@ class RecentlyPlayedSpanTest {
     }
   }
 
-  private fun createRecentlyPlayedActivityIntent(profileId: Int): Intent {
```
I understand removing the `setProfileId` below, but if we also remove this argument, where do we get the profileId to use with the decorator? And why does this test class not have a decorator?

adhiamboperes commented on app/src/main/java/org/oppia/android/app/player/exploration/ExplorationActivity.kt:
```diff
@@ -98,7 +99,6 @@ class ExplorationActivity :
       isCheckpointingEnabled: Boolean
     ): Intent {
       val params = ExplorationActivityParams.newBuilder().apply {
-        this.profileId = profileId
```
There is no corresponding decorator added for this change -- can you clarify if this is because the profileId was actually originally unneccessary? Thad does not seem correct. I suspect that when you call `intent.extractCurrentUserProfileId()`, the default instance is returned. The app does not crash, but some underlying things will definite;y be broken.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/player/exploration/ExplorationFragment.kt:
```diff
@@ -21,15 +20,13 @@ class ExplorationFragment : InjectableFragment() {
       readingTextSize: ReadingTextSize
     ): ExplorationFragment {
       val args = ExplorationFragmentArguments.newBuilder().apply {
-        this.profileId = profileId
```
Whenever a setter is removed, a decorator is expected to be added. Otherwise when retrieved, the profileId will alsways be a default  isntance.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/story/StoryFragmentPresenter.kt:
```diff
@@ -127,8 +128,8 @@ class StoryFragmentPresenter @Inject constructor(
     if (canExplorationBeResumed) {
+      activity.intent.decorateWithUserProfileId(profileId)
       routeToResumeLessonListener.routeToResumeLesson(
-        profileId,
```
I do not think it is corect to remove this, if you follow the usage path, there will be an intent that needs to be decorated with this value.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/story/StoryFragmentPresenter.kt:
```diff
@@ -127,8 +128,8 @@ class StoryFragmentPresenter @Inject constructor(
     explorationCheckpoint: ExplorationCheckpoint
   ) {
     if (canExplorationBeResumed) {
+      activity.intent.decorateWithUserProfileId(profileId)
```
Consequently, this seems out of place here.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/survey/SurveyWelcomeDialogFragmentPresenter.kt:
```diff
@@ -85,8 +86,9 @@ class SurveyWelcomeDialogFragmentPresenter @Inject constructor(
+            activity.intent.decorateWithUserProfileId(profileId)
             val intent =
-              SurveyActivity.createSurveyActivityIntent(activity, profileId, topicId, explorationId)
+              SurveyActivity.createSurveyActivityIntent(activity, topicId, explorationId)
```
```suggestion
val intent =
              SurveyActivity.createSurveyActivityIntent(activity, topicId, explorationId)
             intent.decorateWithUserProfileId(profileId)
```

adhiamboperes commented on app/src/main/java/org/oppia/android/app/survey/SurveyWelcomeDialogFragmentPresenter.kt:
```diff
@@ -85,8 +86,9 @@ class SurveyWelcomeDialogFragmentPresenter @Inject constructor(
           }
           is AsyncResult.Success -> {
             oppiaLogger.d("SurveyWelcomeDialogFragment", "Successfully started a survey session")
+            activity.intent.decorateWithUserProfileId(profileId)
```
```suggestion
```

adhiamboperes commented on app/src/main/java/org/oppia/android/app/topic/lessons/TopicLessonsFragmentPresenter.kt:
```diff
@@ -312,8 +312,8 @@ class TopicLessonsFragmentPresenter @Inject constructor(
             override fun onChanged(it: AsyncResult<ExplorationCheckpoint>) {
               if (it is AsyncResult.Success) {
                 explorationCheckpointLiveData.removeObserver(this)
+                activity.intent.decorateWithUserProfileId(profileId)
```
This is not required, please revert to the original profileId argument.

adhiamboperes replied:
Ditto below.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerFragment.kt:
```diff
@@ -120,10 +117,9 @@ class QuestionPlayerFragment :
+    fun newInstance(readingTextSize: ReadingTextSize):
       QuestionPlayerFragment {
         val args = QuestionPlayerFragmentArguments.newBuilder().apply {
-          this.profileId = profileId
```
Missing the decorator

adhiamboperes commented on app/src/sharedTest/java/org/oppia/android/app/resumelesson/ResumeLessonActivityTest.kt:
```diff
@@ -203,7 +202,6 @@ class ResumeLessonActivityTest {
   private fun createResumeLessonActivityIntent(): Intent {
     return ResumeLessonActivity.createResumeLessonActivityIntent(
       context,
-      ProfileId.newBuilder().apply { internalId = 1 }.build(),
```
This does not need to be removed.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedActivity.kt:
```diff
@@ -61,7 +60,6 @@ class RecentlyPlayedActivity :
   }
 
   override fun routeToExploration(
-    profileId: ProfileId,
```
The profileIds in this class should not be removed.

adhiamboperes commented on app/src/main/java/org/oppia/android/app/home/RouteToExplorationListener.kt:
```diff
@@ -1,12 +1,10 @@
 /** Listener for when an activity should route to a exploration. */
 interface RouteToExplorationListener {
   fun routeToExploration(
-    profileId: ProfileId,
```
This argument is required by the lsitenr implementation.

adhiamboperes reviewed (changes requested):
Thanks @jainv4156! 

I have reviewed the changes -- not all previous comments were addressed, and I have left additional comments that I hpe clarify the expected changes. PTAL.

Additionally, when you need someone to look at your PR, please make sure you assign it to them.
